homeassistant/components/sensor/__init__.py

```diff
@@ -546,6 +546,81 @@ class SensorStateClass(StrEnum):
     SensorDeviceClass.WIND_SPEED: set(UnitOfSpeed),
 }
 
+DEVICE_CLASS_STATE_CLASSES: dict[SensorDeviceClass, set[SensorStateClass | None]] = {
+    SensorDeviceClass.APPARENT_POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.AQI: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ATMOSPHERIC_PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.BATTERY: {None, SensorStateClass.MEASUREMENT},
```

Comment:
Should battery be a `total` instead? If it had a unit that would be Ah aka the integral of the current. So current and battery should be in a similar relation to power (measurement) and energy (total).

Reply:
Battery is expressed in %, this relative, a total isn’t useful (and not allowed/correct, as documented)